### PR TITLE
Adding comments view

### LINKED ISSUE #101: Adding comments

### PR DESCRIPTION
**Pull request by OmerCS8**
- Adding comments to each post in feed view and detail view
- Adding `comment-create` view
- Adding tests
- Changing URL name from `posts-detail` to `post-detail`
Depending on #94 
fix #101
image

### CODE REVIEW

**Review by yulisuliman — DISMISSED** [review]

**Comment by yulisuliman** [comment]
It would be nice to add a picture to PR so we can see the changes :)

**Comment by OmerCS8** [comment]
> It would be nice to add a picture to PR so we can see the changes :)
Done :)

File: posts/urls.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
 from django.urls import path
-from .views import PostListView, PostDetailView
+from .views import PostListView, PostDetailView, CommentCreateView
 from . import views
 
 
 urlpatterns = [
     path('', PostListView.as_view(), name='posts-feed'),
-    path('post/<int:pk>/', PostDetailView.as_view(), name='posts-detail'),
+    path('post/<int:pk>/', PostDetailView.as_view(), name='post-detail'),
```
Comment (TomerNewmanPrograms) [inline posts/urls.py:8]: I think it shouldn't be a part of the PR

**Review by TomerNewmanPrograms — CHANGES_REQUESTED** [review]

File: posts/urls.py (hunk 1 above)
Comment (OmerCS8) [reply posts/urls.py:8]: I am just changing the name of the URL to `post-detail` because every detail belongs to different post and not to all posts :)

**Comment by OmerCS8** [comment]
> - Adding comments to each post in feed view and detail view
> - Adding `comment-create` view
> - Adding tests
> -Changing `posts-detail` URL to `post-detail`
> Depending on #94 
> fix #101
> image
> image
>

File: posts/urls.py (hunk 1 above)
Comment (OmerCS8) [reply posts/urls.py:8]: Added this change to the PR description 👌🏼

**Review by AlonShakaroff — DISMISSED** [review]
LGTM, nice job 👍

**Review by TomerNewmanPrograms — DISMISSED** [review]

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Review by AlonShakaroff — DISMISSED** [review]

**Review by MatanP12 — DISMISSED** [review]

File: posts/templates/posts/feed.html [hunk 2]
```diff
@@ -17,7 +18,24 @@
...
+                <hr class="comment-hr">
+                <p class="comment-content">{{ comment.comment_text }}</p>
+              </div>
+            {% endfor %}
+          </div>
+
         </article>
-        <hr>
+
+    </article>
```
Comment (natifridman) [inline posts/templates/posts/feed.html:39]: I think this is redundant

**Review by natifridman — CHANGES_REQUESTED** [review]

**Review by yulisuliman — APPROVED** [review]

File: posts/templates/posts/feed.html (hunk 2 above)
Comment (OmerCS8) [reply posts/templates/posts/feed.html]: Fixed 👍

**Review by MatanP12 — APPROVED** [review]
LGTM Great Work 🙂

File: posts/tests/test_posts_views.py [hunk 3]
```diff
@@ -35,3 +35,14 @@ def test_resume_creation_view_without_authenticated_user(self, client):
...
+    def test_comment_creation_view_with_authenticated_user(self, client, persist_user, persist_resume):
+        client.force_login(persist_user)
+        comment_creation_path = f'/post/{persist_resume.pk}/comment/'
+        assert client.get(comment_creation_path).status_code == 200
+        client.logout()
+
+    def test_comment_creation_view_without_authenticated_user(self, client, persist_resume):
+        comment_creation_path = f'/post/{persist_resume.pk}/comment/'
+        response = client.get(comment_creation_path)
+        assert response.status_code == 302
```
Comment (AlonShakaroff) [inline posts/tests/test_posts_views.py:48]: same here.

**Review by AlonShakaroff — CHANGES_REQUESTED** [review]

File: posts/tests/test_posts_views.py [hunk 4]
```diff
@@ -35,3 +35,14 @@ def test_resume_creation_view_without_authenticated_user(self, client):
         resume_creation_path = urls.reverse('resume-create')
         response = client.get(resume_creation_path)
         assert response.status_code == 302
+
+    def test_comment_creation_view_with_authenticated_user(self, client, persist_user, persist_resume):
+        client.force_login(persist_user)
+        comment_creation_path = f'/post/{persist_resume.pk}/comment/'
+        assert client.get(comment_creation_path).status_code == 200
```
[the remark this answers is not in the document]
Comment (OmerCS8) [reply posts/tests/test_posts_views.py]: Done

File: posts/tests/test_posts_views.py (hunk 3 above)
Comment (OmerCS8) [reply posts/tests/test_posts_views.py]: Done

**Review by AlonShakaroff — DISMISSED** [review]
LGTM, look really nice

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by yulisuliman — CHANGES_REQUESTED** [review]
You have to resolve conflicts.
besides that LGTM :)
@OmerCS8

**Review by yulisuliman — DISMISSED** [review]

**Review by natifridman — COMMENTED** [review]
Maybe we can add some more tests? For example, what happens when you use a long comment or an empty one?

**Comment by OmerCS8** [comment]
> Maybe we can add some more tests? For example, what happens when you use a long comment or an empty one?

Don't we already test scenarios of this kind in the `test_comment` file?
Do you mean to test that we stay in the `comment_form` template if we make an unvalid comment?

**Comment by natifridman** [comment]
> > Maybe we can add some more tests? For example, what happens when you use a long comment or an empty one?
> 
> Don't we already test scenarios of this kind in the `test_comment` file? Do you mean to test that we stay in the `comment_form` template if we make an unvalid comment?

yes

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Comment by OmerCS8** [comment]
> > > Maybe we can add some more tests? For example, what happens when you use a long comment or an empty one?
> > 
> > 
> > Don't we already test scenarios of this kind in the `test_comment` file? Do you mean to test that we stay in the `comment_form` template if we make an unvalid comment?
> 
> yes

Done 👍

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]
image

**Review by natifridman — APPROVED** [review]

**Comment by OmerCS8** [comment]
All the commits are squashed and ready to merge😁

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by yulisuliman — APPROVED** [review]
